app/controllers/logs_controller.rb

```diff
@@ -41,7 +41,14 @@ def update
   end
 
   def show
-    redirect_to edit_log_path(@log) unless @log.activated?
+    if @log.activated?
+      user = User.find(@log.user_id)
+      user_logs = user.logs.active.latest.reverse
+      @next_log = user_logs[user_logs.index(@log)+1] if user_logs.index(@log)+1 != user_logs.size
```

Comment:
I think it might be better to use `<` rather than `!=` because it will keep from allowing something to be equal and will prevent `next_log` from ever being greater than `user_logs.size`

Also, it seems like it works this way, but my brain is wondering how the `user_logs.size` stays correct when the index starts at 0. I keep thinking like it needs to be `user_logs.size - 1`, but this still works. Not a problem to be solved, but what am I missing? 😂

Reply:
@dhuey Yeah the logic took a little while for me to understand. So the range goes from 0..user_logs.size - 1. Now the check is to see if the next one (index of current log + 1) != user_logs.size, which is equivalent to saying the index of the current log != user_logs.size - 1